new-log-viewer/src/components/StatusBar/LogLevelSelect/index.tsx

```diff
@@ -0,0 +1,273 @@
+import KeyboardArrowUpIcon from "@mui/icons-material/KeyboardArrowUp";
+import RemoveIcon from "@mui/icons-material/Remove";
+
+import {
+    LOG_LEVEL,
+    LOG_LEVEL_NAMES,
+    MAX_LOG_LEVEL,
+} from "../../../typings/logs";
+import {range} from "../../../utils/data";
+import LogLevelChip from "./LogLevelChip";
+
+import "./index.css";
+
+
+/**
+ * Renders a `<Tooltip/>` with its placement set to the left of the children element.
+ *
+ * @param props
+ * @param props.children
+ * @param props.rest
+ * @return
+ */
+const PlacementLeftTooltip = ({children, ...rest}: TooltipProps) => (
+    <Tooltip
+        {...rest}
+        placement={"left"}
+    >
+        {children}
+    </Tooltip>
+);
+
+/**
+ * Renders a `<Tooltip/>` with its placement set to the right of the children element.
+ *
+ * @param props
+ * @param props.children
+ * @param props.rest
+ * @return
+ */
+const PlacementRightTooltip = ({children, ...rest}: TooltipProps) => (
+    <Tooltip
+        {...rest}
+        placement={"right"}
+    >
+        {children}
+    </Tooltip>
+);
+
+interface LogSelectOptionProps {
+    isChecked: boolean,
+    logLevelName: string,
+    logLevelValue: LOG_LEVEL,
+    onCheckboxClick: React.MouseEventHandler
+}
+
+/**
+ * Renders an <Option/> in the <LogLevelSelect/> for selecting some log level and the level above
+ * it.
+ *
+ * @param props
+ * @param props.isChecked
+ * @param props.logLevelName
+ * @param props.logLevelValue
+ * @param props.onCheckboxClick
+ * @return
+ */
+const LogSelectOption = ({
+    isChecked,
+    logLevelName,
+    logLevelValue,
+    onCheckboxClick,
+}:LogSelectOptionProps) => {
+    return (
+        <Option
+            data-value={logLevelValue}
+            key={logLevelName}
+            value={logLevelValue}
+        >
+            <ListItemDecorator>
+                <PlacementLeftTooltip
+                    title={
+                        <Stack
+                            alignItems={"center"}
+                            direction={"row"}
+                        >
+                            {isChecked ?
+                                <RemoveIcon/> :
+                                <AddIcon/>}
+                            {logLevelName}
+                        </Stack>
+                    }
+                >
+                    <Checkbox
+                        checked={isChecked}
+                        size={"sm"}
+                        value={logLevelValue}
+                        onClick={onCheckboxClick}/>
+                </PlacementLeftTooltip>
+            </ListItemDecorator>
+            <PlacementRightTooltip
+                title={
+                    <Stack
+                        alignItems={"center"}
+                        direction={"row"}
+                    >
+                        {logLevelName}
+                        {" and below"}
+                    </Stack>
+                }
+            >
+                <ListItemContent data-value={logLevelValue}>
```

Comment:
For my own understanding, could you explain why we need data-value on both the Option and this?

Reply:
When `onCheckboxClick(ev)` is invoked, the event target can either be the Option (an `<ul/>`) or the ListItemContent (a `<div/>` as a child of the <ul/>`) depending on whether the user is clicking the edges of the option.

Having the data attribute on both elements allows the handler to read the value without too many changes in the handler itself.

Reply:
Hm, I guess this is because we can't use currentTarget, right? If so, do you think it's worth adding a comment?

Reply:
> this is because we can't use currentTarget

With the handler attached to the `<Select/>`, no. However, there is no real benefit of passing the handler to the `onChange` prop of `<Select/>` now we are not even utilizing the second parameter (newValue) of `onChange`. 

With the handler attached to the `<Option/>` directly, yes. I'll make the change.